### PR TITLE
Add application branch for alumni mentors

### PR DESCRIPTION
**Pull request by bunsenmcdubbs**
This PR contains the application branch to enable the applications for alumni mentors as detailed in https://docs.google.com/document/d/1WZng68MfXSQTj9J290civ5_fIc_qGBtVbmc2aHCzQ70/edit

cc @faizanv 

my local dev environment is broken (partially because of #132) - pushing here to see if the PR can run on the cluster.

Dear fam,
Please proofread me.
❤️,
Andrew

### CODE REVIEW

File: server/config/questions.json [hunk 1]
```diff
@@ -1118,6 +1118,356 @@
...
+            },
+            {
+                "name": "why-do-you-want-to-mentor",
+                "label": "Why do you want to mentor?",
+                "type": "textarea",
+                "required": true
+            },
+            {
+                "name": "other",
+                "label": "Anything else? Drop us some links (Github, LinkedIn, Personal Site etc.)",
```
Comment (petschekr) [inline server/config/questions.json]: Correct capitalization (GitHub, LinkedIn, personal site)

File: server/config/questions.json [hunk 2]
```diff
@@ -1118,6 +1118,356 @@
...
+                    "Wright State University",
+                    "Yale University",
+                    "Yeshiva University"
+                ],
+                "hasOther": true,
+                "required": true
+            },
+            {
+                "name": "graduation-year",
+                "label": "Graduation Year",
```
Comment (petschekr) [inline server/config/questions.json]: Graduation year to be more consistent

File: server/config/questions.json (hunk 1 above)
Comment (petschekr) [reply server/config/questions.json]: Also comma before "etc."

**Comment by petschekr** [comment]
Just rebased this to fix some awful merging that happened at our meeting. Please `git reset --hard origin/alumni-mentor-application` if you want to add commits

**Comment by faizanv** [comment]
Hmm seems like there has been an overlap. Please incorporate the changes to the questions json I just made on my seemingly duplicate PR to this one @petschekr

**Pull request merged**